app/tests/unit/lib/theme/widgets/SelectParentWidget.test.tsx

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import SelectParentWidget from "lib/theme/widgets/SelectParentWidget";
+
+import { render, fireEvent, screen } from "@testing-library/react";
+
+const props: any = {
+  id: "test-id",
+  onChange: jest.fn(),
+  value: undefined,
+  required: false,
+  parent: {
+    list: [
+      {
+        rowId: 1,
+        name: "EP",
+        fundingStreamId: 1,
+      },
+      {
+        rowId: 2,
+        name: "IA",
+        fundingStreamId: 2,
+      },
+    ],
+    label: "Parent Label",
+    placeholder: "parent placeholder",
+    displayField: "name",
+  },
+  child: {
+    list: [
+      {
+        rowId: 1,
+        year: 2019,
+        fundingStreamId: 1,
+      },
+      {
+        rowId: 2,
+        year: 2020,
+        fundingStreamId: 1,
+      },
+      {
+        rowId: 3,
+        year: 2021,
+        fundingStreamId: 2,
+      },
+      {
+        rowId: 4,
+        year: 2022,
+        fundingStreamId: 2,
+      },
+    ],
+    label: "Child Label",
+    placeholder: "child placeholder",
+    displayField: "year",
+  },
+  foreignKey: "fundingStreamId",
+};
+
+describe("The SelectParentWidget Widget", () => {
+  it("selects the first option from parent", async () => {
+    const selectParentWidget = render(<SelectParentWidget {...props} />);
+    const parentDropdown = selectParentWidget.container.querySelector(
+      "#select-parent-dropdown-test-id"
+    );
```

Comment:
either way works. The id comes from the actual component so using querySelector gives you slightly more code coverage.

Reply:
I think it comes down to the philosophy behind `testing-library`. Their recommendation is to only test based on things that the user can see, i.e. in this case accessing the input via its label instead of the id.